## 0. Load Library

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

## 1. What is the dimension (col, row) of the data frame?

In [21]:
#Melakukan loading data
df = pd.read_csv('titanic.csv')

In [22]:
#Melihahat 5 data teratas untuk checking
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [23]:
#Melihat dimensi dari data
df.shape

(887, 8)

- Dimensi data dari Titanic memiliki 887 Rows dan 8 Columns

## 2. How to know data type of each variable?

In [24]:
#Untuk mengetahui tipe data dari setiap features
df.dtypes

Survived                     int64
Pclass                       int64
Name                        object
Sex                         object
Age                        float64
Siblings/Spouses Aboard      int64
Parents/Children Aboard      int64
Fare                       float64
dtype: object

- Data titanic memiliki 4 feature yang tipe datanya integer, 2 yang float, dan 2 yang string.

## 3. How many passengers survived (Survived=1) and not-survived (Survived=0)?

In [25]:
#Grouping data berdasarkan feature survived agar mengetahui jumlah passenger yang selamat dan tidak selamat
df.groupby('Survived').size()

Survived
0    545
1    342
dtype: int64

- Data titanic memperlihatkan bahwa terdapat 342 penumpang yang selamat dan 545 penumpang yang tidak selamat.

## 4. How to drop column ‘Name’ from the data frame?

In [26]:
#Data sebelum features name di drop
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [8]:
#Dropping features name
df.drop(columns=['Name'], inplace=True)

In [9]:
#data setelah feature name di drop
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  
0                        0   7.2500  
1                        0  71.2833  
2                        0   7.9250  
3                        0  53.1000  
4                        0   8.0500

## 5. Add one new column called ‘family’ to represent number of family-member aboard (hint: family = sibsp + parch)

In [27]:
#membuat feature baru bernama family
#pertambahan antara feature siblings/spouses aboard dan feature parents/children aboard
df['family'] = df['Siblings/Spouses Aboard'] + df['Parents/Children Aboard']
df.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses Aboard  Parents/Children Aboard     Fare  \
0    male  22.0                        1                        0   7.2500   
1  female  38.0                        1                        0  71.2833   
2  female  26.0                        0                        0   7.9250   
3  female  35.0                        1                        0  53.1000   
4    male  35.0                        0                        0   8.0500   

   family  
0       1  
1       1  
2       0  
3       1  
4       0

## 6. As shown, columns ‘Age’ contains missing values. Please add new column named ‘Age_miss’ to indicate whether Age is missing or not (Age_miss = ‘YES’ for missing value and ‘NO’ for non-missing value). 


In [11]:
#checking data apakah ada missing values apa tidak
#tidak ada missing values
df.isnull().sum()

Survived                   0
Pclass                     0
Sex                        0
Age                        0
Siblings/Spouses Aboard    0
Parents/Children Aboard    0
Fare                       0
family                     0
dtype: int64

In [12]:
#malukan looping untuk membuat feature baru bernama age_miss
#nilai yes menandakan bahwa kolom tersebut age nya missing dan nilai NO sebaliknya
age_miss = []
for i in range(len(df)):
    if df['Age'][i] == 0:
        age_miss.append('Yes')
    else:
        age_miss.append('No')

In [13]:
#memasukan feature age_miss ke dalam df
df['age_miss'] = age_miss
df.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0         0       3    male  22.0                        1   
1         1       1  female  38.0                        1   
2         1       3  female  26.0                        0   
3         1       1  female  35.0                        1   
4         0       3    male  35.0                        0   

   Parents/Children Aboard     Fare  family age_miss  
0                        0   7.2500       1       No  
1                        0  71.2833       1       No  
2                        0   7.9250       0       No  
3                        0  53.1000       1       No  
4                        0   8.0500       0       No

## 7. Please fill Age missing value with means of existing Age values

In [14]:
#mengisi data yang kosong di feature name dengan rata-rata age
df['Age'] = df['Age'].fillna(df['Age'].mean())

## 8. What is the maximum passenger Age who survived from the tragedy? 

In [15]:
df[df['Survived']==1]['Age'].max()

80.0

- Umur passenger paling tinggi yang selamat adalah 80 tahun.

## 9. How many passengers survived from each ‘PClass’? 

In [16]:
df[df['Survived']==1].groupby('Pclass')['Survived'].count()

Pclass
1    136
2     87
3    119
Name: Survived, dtype: int64

- Kalau dilihat dari Pclassnya maka:
    - Passenger yang selamat dengan Pclass 1 adalah 136
    - Passenger yang selamat dengan Pclass 2 adalah 87
    - Passenger yang selamat dengan Pclass 3 adalah 119

## 10. How to randomly split the data frame into 2 parts (titanic1 and titanic2) with proportion of 0.7 for tttanic1 and 0.3 for titanic2 ? 


- Pertama membuat dataframe titanic 1 dengan proporsi 70%

In [17]:
titanic1 = df.sample(frac=0.7).reset_index()
titanic1.head()

index  Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
0    606         1       1  female  40.0                        0   
1    830         0       3    male  18.0                        0   
2    381         1       1  female  35.0                        1   
3    580         0       1    male  36.0                        0   
4     57         1       2  female   5.0                        1   

   Parents/Children Aboard      Fare  family age_miss  
0                        0  153.4625       0       No  
1                        0    8.3000       0       No  
2                        0   52.0000       1       No  
3                        0   40.1250       0       No  
4                        2   27.7500       3       No

In [18]:
titanic1.shape

(621, 10)

- Lalu membuat dataframe titanic2 dengan proporsi 30%

In [19]:
titanic2 = df.drop(titanic1.index)
titanic2.head()

Survived  Pclass     Sex   Age  Siblings/Spouses Aboard  \
621         0       3    male  21.0                        0   
622         0       1    male  61.0                        0   
623         0       2    male  57.0                        0   
624         1       1  female  21.0                        0   
625         0       3    male  26.0                        0   

     Parents/Children Aboard     Fare  family age_miss  
621                        0  16.1000       0       No  
622                        0  32.3208       0       No  
623                        0  12.3500       0       No  
624                        0  77.9583       0       No  
625                        0   7.8958       0       No

In [20]:
#Featuresnya 9 karena tidak ada kolom index
titanic2.shape

(266, 9)